protopipe/scripts/write_dl2.py

```diff
@@ -389,39 +460,87 @@ class RecoEvent(tb.IsDescription):
-                    # this should be read in some way from
-                    # the classifier configuration file!!!!!
-
-                    features_img = np.array(
-                        [
-                            np.log10(reco_energy),
-                            np.log10(energy_tel_classifier[tel_id]),
-                            np.log10(moments.intensity),
-                            moments.width.value,
-                            moments.length.value,
-                            h_max.value,
-                            impact_dict[tel_id].value,
-                        ]
-                    )
+
+                    ############################################################
+                    #                  GET FEATURES
+                    ############################################################
+
+                    # Read feature list from model configutation file
+                    features_basic = classifier_config["FeatureList"]["Basic"]
+                    features_derived = classifier_config["FeatureList"]["Derived"]
+                    features = features_basic + list(features_derived)
+
+                    # Create a pandas Dataframe with basic quantities
+                    # This is needed in order to connect the I/O system of the
+                    # model inputs to the in-memory computation of this script
+                    data = pd.DataFrame({
+                        "hillas_intensity": [moments.intensity],
+                        "hillas_width": [moments.width.to("deg").value],
+                        "hillas_length": [moments.length.to("deg").value],
+                        "hillas_x": [moments.x.to("deg").value],
+                        "hillas_y": [moments.y.to("deg").value],
+                        "hillas_phi": [moments.phi.to("deg").value],
+                        "hillas_r": [moments.r.to("deg").value],
+                        "leakage_intensity_width_1_reco": [leakage_dict[tel_id]['leak1_reco']],
+                        "leakage_intensity_width_2_reco": [leakage_dict[tel_id]['leak2_reco']],
+                        "leakage_intensity_width_1": [leakage_dict[tel_id]['leak1']],
+                        "leakage_intensity_width_2": [leakage_dict[tel_id]['leak2']],
+                        "az": [reco_result.az.to("deg").value],
+                        "alt": [reco_result.alt.to("deg").value],
+                        "h_max": [h_max.value],
+                        "impact_dist": [impact_dict[tel_id].to("m").value],
+                        "reco_energy": reco_energy,
+                        "reco_energy_tel": energy_tel_classifier[tel_id],
+                    })
+
+                    # Compute derived features and add them to the dataframe
+                    for key, expression in features_derived.items():
+                        if key not in data:
+                            data.eval(f'{key} = {expression}', inplace=True)
+
+                    print(data)
```

Comment:
do you really want to print this every time?

Reply:
nope, leftover!